course_content/notebooks/dask_intro.ipynb

```diff
@@ -831,6 +831,94 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First we should define what we mean when we talk about `map` and `filter`:\n",
+    "\n",
+    "`map(func, seq)`: A function which takes a function (`func`) and a sequence of inputs (`seq`) as arguments. It returns a sequence of outputs which are the result of applying `func` to each of the items in `seq`.\n",
+    "\n",
+    "`filter(func, seq)`: A function which also takes a function (`func`) and a sequence of inputs (`seq`) as arguments. `func` must be a function which returns a Boolean value, i.e. either `True` or `False`. This function will be applied to every element of `seq` and only if `func` returns `True` will the element of the sequence be included in the output sequence."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`filter` sounds very similar to what we were doing when using `iris.load` with a `constraint`. In a similar way to how we generated `bag_o_cubes` we can generate a `dask.bag` by mapping a sequence of filenames onto `iris.load`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "bag_of_cubes = db.from_sequence(files).map(iris.load)\n",
+    "bag_of_cubes.compute()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can use the `filter` function of a `bag_of_cubes` to constrain it based on the phenomenon it represents. To do this we need to define a filter function:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "phenom_filter = lambda cube: cube.name() == 'air_temperature'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this case we used a `lambda` to define our filter function, which returns `True` if the name of the phenomenon represented by the cube is `'air_temperature'`.\n",
+    "\n",
+    "`bag_of_cubes` is a nested list of lists, with each list containing the cubes from each file. We can use `flatten` to reduce it to a single list of cubes then apply `filter` to select cubes based on a phenomenon:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "filtered_bag = bag_of_cubes.flatten().filter(phenom_filter)\n",
+    "filtered_bag.compute()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Again we can use the magic of Python to chain this all into one line of code:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "filtered_cubes = db.from_sequence(files).map(iris.load).flatten().filter(lambda c: c.name()=='surface_altitude').compute()"
```

Comment:
Phew, this line is _far_ too long - it would never fit into the 80char Python line limit length 😱 To improve this, you could consider...
* shortening the variable name
* using the filtering function `phenom_filter` that you define above
* calling compute separately